tests/unittests/parsing/VisitorTests.cpp

```diff
@@ -733,6 +736,65 @@ class C; endclass
         }
     }
 }
+TEST_CASE("SyntaxTree/Compilation Invariant Checking") {
+
+    // Validates that the parent pointers in the syntax tree are correct, and provides debug info if
+    // not
+    auto validateParents = [](const syntax::SyntaxTree& tree) {
+        bool valid = true;
+        tree.root().visit(makeAllSyntaxVisitor([&](const SyntaxNode& node) {
+            if (node.kind == SyntaxKind::SyntaxList || node.kind == SyntaxKind::SeparatedList) {
+                return;
+            }
+            for (size_t i = 0; i < node.getChildCount(); i++) {
+                auto child = node.childNode(i);
+                if (!child)
+                    continue;
+                if (child->parent != &node) {
+                    valid = false;
+                    auto parentKind = toString(child->parent->kind);
+                    auto childKind = toString(child->kind);
+                    auto expectedParent = toString(node.kind);
+                    fmt::print("Parent pointer mismatch with `{}`. {}.parent should be {}, "
```

Comment:
Prefer using catch2's `INFO` macro so this gets categorized under the correct test case.